CONTRIBUTION-GUIDE.md

```diff
@@ -0,0 +1,151 @@
+
+- **`create-nil-hardhat-project/`** – Provides a template or scaffolding for developers to quickly set up a Hardhat project tailored for the nil blockchain, streamlining the development and deployment of smart contracts.
+
+- **`docs/`** – Contains comprehensive documentation, including protocol specifications, developer guides, API references, and integration tutorials.
+
+- **`explorer_backend/`** – Manages the backend services for the blockchain explorer, including data indexing, API endpoints, and real-time data processing.
+
+- **`explorer_frontend/`** – Develops the user interface for the blockchain explorer, enabling users to visualize blockchain data, track transactions, and monitor network status.
+
+- **`niljs/`** – A TypeScript client library providing tools and abstractions for interacting with the nil blockchain.
+
+- **`rpc-types/`** – Defines TypeScript types and interfaces for the Remote Procedure Call (RPC) methods used by the nil blockchain.
+
+- **`smart-contracts/`** – Hosts the `@nilfoundation/smart-contracts` package, comprising essential smart contracts for blockchain interactions.
+
+(_Note: Some directories like **`nix`** are omitted for brevity, but contributors are encouraged to explore the full repository if necessary._)
+
+## How to Contribute
+
+This section describes in some detail how changes can be made and proposed with pull requests.
+
+### 1. Check Existing Issues
+
+Before starting work on a feature or bug fix, check existing issues. If none exist, consider opening an issue first to discuss your approach. This ensures alignment with project maintainers and avoids duplicate work.
+
+### 2. Fork the Repository
+
+- Click the **Fork** button on the repository's page.
+- Clone your forked repository to your local machine:
+  ```sh
+  git clone https://github.com/your-username/nil.git
+  ```
+- Navigate to the project directory:
+  ```sh
+  cd nil
+  ```
+
+### 3. Create a Branch
+
+- Create a new branch for your feature or bug fix:
+  ```sh
+  git checkout -b feature-branch-name
+  ```
+
+### 4. Make Changes
+
+- Implement your changes following the project’s coding style.
+- Ensure that all functional changes include appropriate tests.
+- Test your changes thoroughly before submitting.
+- Commit and sign your changes with a descriptive message:
+  ```sh
+  git add .
+  git commit -S -m "Add feature or fix description"
+  ```
+
+### 5. Push and Create a Pull Request
+
+- Push your changes to your forked repository:
+  ```sh
+  git push origin feature-branch-name
+  ```
+- Open a pull request from your forked repository to the main repository.
+- Provide a clear description of your changes and the motivation behind them.
+
+## Contribution Guidelines
+
+### PR Review and Merging Process
+
+- **PRs will not be merged directly**. Instead, maintainers will review and rebase them before merging.
```

Comment:
This santance is not clear for me. I would phrase like this
PRs require approval and will be merged by a team member after signing, not directly by the author